pkg/server/container_create_windows.go

```diff
@@ -527,39 +527,70 @@ func (c *criService) generateContainerSpec(id string, sandboxID string, sandboxP
 		g.SetProcessUsername(username)
 	}
 
-	// For both LCOW and WCOW, devices are passed through from the host to the
-	// container via the OCI Windows.Devices field
-	if err := addOCIWindowsDevices(&g, config.Devices); err != nil {
+	if err := addOCIDevices(&g, plat.OS, config.Devices); err != nil {
 		return nil, err
 	}
 
 	return g.Config, nil
 }
 
-// getWindowsDeviceInfo is a helper function that returns a spec specified device's
+// getCRIDeviceInfo is a helper function that returns a spec specified device's
 // prefix and device identifier. A prefix is any string before "://" in the spec
 // device's `HostPath`.
-func getWindowsDeviceInfo(hostPath string) (string, string, error) {
+func getCRIDeviceInfo(hostPath string) (string, string, error) {
 	substrings := strings.SplitN(hostPath, "://", 2)
 	if len(substrings) <= 1 {
 		return "", "", fmt.Errorf("failed to parse device information for %s", hostPath)
 	}
 	return substrings[0], substrings[1], nil
 }
 
-// addOCIWindowsDevices parses the devices field on the spec and creates corresponding
-// `WindowsDevice` on the container config if valid.
-func addOCIWindowsDevices(g *generator, devs []*runtime.Device) error {
-	for _, device := range devs {
-		prefix, deviceIdentifier, err := getWindowsDeviceInfo(device.HostPath)
-		if err != nil {
-			return err
-		}
-		device := runtimespec.WindowsDevice{
-			ID:     deviceIdentifier,
-			IDType: prefix,
+// hasWindowsDevicePrefix returns true if the device's hostPath contains a prefix.
+// A prefix is any string before "://" in the spec device's `HostPath`.
+func hasWindowsDevicePrefix(hostPath string) bool {
+	substrings := strings.SplitN(hostPath, "://", 2)
+	if len(substrings) <= 1 {
+		return false
+	}
+	return true
+}
+
+// addOCIDevices converts cri spec devices to `LinuxDevice`s or `WindowsDevice`s
+//
+// If the device's `HostPath` contains a prefix as returned by a call to `getCRIDeviceInfo`,
+// convert the cri device to a `WindowsDevice`. Otherwise, if we're running WCOW, convert
+// the cri device to a `WindowsDevice` with an unknown `IDType`. If we're running LCOW,
+// convert the cri device to a skeleton `LinuxDevice` where the remaining fields
+// must be filled out in the GCS.
+func addOCIDevices(g *generator, OS string, devices []*runtime.Device) error {
+	for _, dev := range devices {
+		if hasWindowsDevicePrefix(dev.HostPath) {
+			// handle as a WindowsDevice with known type
+			prefix, identifier, err := getCRIDeviceInfo(dev.HostPath)
+			if err != nil {
+				return err
+			}
+			device := runtimespec.WindowsDevice{
+				ID:     identifier,
+				IDType: prefix,
+			}
+			g.Config.Windows.Devices = append(g.Config.Windows.Devices, device)
+		} else if OS == "windows" {
+			// handle as a WindowsDevice with unknown type
+			device := runtimespec.WindowsDevice{
+				ID:     dev.HostPath,
+				IDType: "unknown",
```

Comment:
Do we have a use case for this "unknown" device type today?

Reply:
We don't. The intention here was to be complete with linux vs windows options here, but can be removed.